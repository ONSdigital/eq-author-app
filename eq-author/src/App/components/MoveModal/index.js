import React from "react";
import Modal from "components/Modals/Modal";
import styled from "styled-components";
import DialogHeader from "components/Dialog/DialogHeader";
import { Message, Heading } from "components/Dialog/DialogMessage";
import PropTypes from "prop-types";
import { colors } from "constants/theme";

const StyledModal = styled(Modal)`
  .Modal {
    width: 25em;
  }
`;

const CenteredHeading = styled(Heading)`
  text-align: center;
  margin-bottom: 1rem;
  color: ${colors.text};
`;

const MoveModal = ({ title, children, isOpen, onClose }) => {
  return (
    <StyledModal
      isOpen={isOpen}
      onClose={onClose}
      data={{ test: "move-modal" }}
    >
      <DialogHeader>
        <Message>
          <CenteredHeading>{title}</CenteredHeading>
        </Message>
      </DialogHeader>
      {children}
    </StyledModal>
  );
};

MoveModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default MoveModal;
