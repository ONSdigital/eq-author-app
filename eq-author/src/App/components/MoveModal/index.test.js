import React from "react";
import { shallow } from "enzyme";
import MoveModal from "App/components/MoveModal";

const createWrapper = (props = {}, render = shallow, children = undefined) => {
  return render(<MoveModal {...props}>{children}</MoveModal>);
};

describe("components/MoveModal", () => {
  let props;
  let wrapper;
  let children;

  beforeEach(() => {
    props = {
      isOpen: true,
      onClose: jest.fn(),
      title: "Modal title"
    };

    children = <p>Modal content</p>;

    wrapper = createWrapper(props, shallow, children);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
