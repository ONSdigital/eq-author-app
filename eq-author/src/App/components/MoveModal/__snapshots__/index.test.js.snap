// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MoveModal should render 1`] = `
<MoveModal__StyledModal
  data={
    Object {
      "test": "move-modal",
    }
  }
  isOpen={true}
  onClose={[MockFunction]}
>
  <DialogHeader___default>
    <DialogMessage__Message>
      <MoveModal__CenteredHeading>
        Modal title
      </MoveModal__CenteredHeading>
    </DialogMessage__Message>
  </DialogHeader___default>
  <p>
    Modal content
  </p>
</MoveModal__StyledModal>
`;
